src/main/java/frc/robot/poseestimator/PoseEstimatorMath.java

```diff
@@ -0,0 +1,94 @@
+		boolean hasGyroUpdated = currentGyroAngle != null;
+		if (hasGyroUpdated) {
+			return updateChangeInAngle(twist, currentGyroAngle, lastGyroAngle);
+		}
+		return twist;
+	}
+
+	public static Twist2d updateChangeInAngle(Twist2d twist, Rotation2d currentGyroAngle, Rotation2d lastGyroAngle) {
+		double rotationDifference = currentGyroAngle.getRadians() - lastGyroAngle.getRadians();
+		double wrappedRotationDifference = MathUtil.angleModulus(rotationDifference);
+		return new Twist2d(twist.dx, twist.dy, wrappedRotationDifference);
+	}
+
+	public static double[] combineStandardDeviations(double[] odometryStandardDeviations, double[] visionStandardDeviations) {
+		double[] combinedStandardDeviations = new double[3];
+		for (int i = 0; i < combinedStandardDeviations.length; i++) {
+			double odometryStandardDeviation = odometryStandardDeviations[i];
+			double visionStandardDeviation = visionStandardDeviations[i];
+			combinedStandardDeviations[i] = combineStandardDeviations(odometryStandardDeviation, visionStandardDeviation);
+		}
+		return combinedStandardDeviations;
+	}
+
+	public static double combineStandardDeviations(double odometryStandardDeviation, double visionStandardDeviation) {
+		if (odometryStandardDeviation == 0) {
+			return 0;
+		}
+		double squaredVisionStandardDeviation = visionStandardDeviation * visionStandardDeviation;
+		return odometryStandardDeviation
+			/ (odometryStandardDeviation + Math.sqrt(odometryStandardDeviation * squaredVisionStandardDeviation));
+	}
+
+	public static double[] multiplyArrays(double[] array1, double[] array2) {
+		double[] result = new double[Math.min(array1.length, array2.length)];
+		for (int i = 0; i < result.length; i++) {
+			result[i] = array1[i] * array2[i];
+		}
+		return result;
+	}
+
+	public static Transform2d
+		useKalmanOnTransform(VisionObservation observation, Pose2d currentPoseEstimation, double[] odometryStandardDeviations) {
+		double[] combinedStandardDeviations = PoseEstimatorMath
+			.combineStandardDeviations(observation.standardDeviations(), odometryStandardDeviations);
+		Transform2d visionDifferenceFromOdometry = new Transform2d(currentPoseEstimation, observation.visionPose());
+		return scaleDifferenceFromKalman(visionDifferenceFromOdometry, combinedStandardDeviations);
+	}
+
+	public static Transform2d
+		scaleDifferenceFromKalman(Transform2d visionDifferenceFromOdometry, double[] combinedStandardDeviations) {
+		double[] visionDifferenceFromOdometryMatrix = {
+			visionDifferenceFromOdometry.getX(),
+			visionDifferenceFromOdometry.getY(),
+			visionDifferenceFromOdometry.getRotation().getRadians()};
+		double[] standardDeviationsAppliedTransform = multiplyArrays(
+			combinedStandardDeviations,
+			visionDifferenceFromOdometryMatrix
+		);
+		return new Transform2d(
+			standardDeviationsAppliedTransform[0],
+			standardDeviationsAppliedTransform[1],
+			Rotation2d.fromRadians(standardDeviationsAppliedTransform[2])
```

Comment:
If it's not a record, at least make an enum/consts for each position in the array, like PoseArrayEntryValue (or smth).